src/auth/AuthProvider.tsx

```diff
@@ -2,35 +2,45 @@ import { useMutation } from "@apollo/client";
 import { useState, createContext, useContext, useEffect, FC } from "react";
 import { useNavigate } from "react-router-dom";
 
-import { LOGIN, Login, User } from "../components/pages/LoginPage/interface";
+import { LOGIN, Login } from "../components/pages/LoginPage/interface";
 import { AuthContextType, AuthProps } from "./interface";
+import { getUserDataFromStorage, setUserDataInStorage } from "./utils";
 
 const AuthContext = createContext<AuthContextType>({} as AuthContextType);
+const oneDayInMS = 86400000;
+const oneMonthInMS = 2629800000;
 
 export const AuthProvider: FC<AuthProps> = ({ children }) => {
-  const [user, setUser] = useState<User>();
+  const [user, setUser] = useState<{ token: string; expiry: number }>();
 
   const navigate = useNavigate();
-
-  const userData = localStorage.getItem("user");
-  const parsedUserData = userData && JSON.parse(userData);
+  const userData = getUserDataFromStorage("user");
+  const noUserData = !!userData;
```

Comment:
Here you are making a boolean value out of the object, so no negation. ;) It's rather like creating a flag:
```suggestion
  const isUserDataPresent = !!userData;
```